src/functionalTest/java/submitBulk/IssueWarrant.java

```diff
@@ -0,0 +1,179 @@
+        return xmlContent;
+    }
+
+    @Before
+    public void setUp() {
+        this.xml = soapRequestBody();
+        this.response = soapResponse(xml);
+    }
+
+    @Step("When we send the soap request and get the response")
+    public Response soapResponse(String xml) {
+        Response response = RestAssured
+                .given()
+                .baseUri("http://civil-sdt-demo.service.core-compute-demo.internal")
+                .basePath("/producers/service/sdtapi")
+                .header("Content-Type", "application/soap+xml;charset=UTF-8")
+                .body(xml)
+                .post();
+        return response;
+    }
+
+    @Step("Then the status code should be 200")
+    public void checkStatusCode(Response response) {
+        response.then().statusCode(200);
+        response.prettyPrint();
+    }
+
+    @Test
+    public void issueWarrantResponseCode() {
+
+        XmlPath xmlPath = new XmlPath(XMLFunctions.getCleanResponse(response)).setRoot("Envelope.Body.bulkResponse");
+        String xmlResponseCode = xmlPath.getString("status.@code");
+        String customerReference = xmlPath.getString("customerReference");
+        String sdtBulkReference = xmlPath.getString("sdtBulkReference");
+        String submittedDate = xmlPath.getString("submittedDate");
+        String sdtService = xmlPath.getString("sdtService");
+        String requestCount = xmlPath.getString("requestCount");
+        int statusCode = response.getStatusCode();
+
+        System.out.println("xmlResponseCode: " + xmlResponseCode);
+        System.out.println("Customer Reference: " + customerReference);
+        System.out.println("SDT Bulk Reference: " + sdtBulkReference);
+        System.out.println("Submitted Date: " + submittedDate);
+        System.out.println("SDT Service: " + sdtService);
+        System.out.println("Request Count: " + requestCount);
+
+        assertThat(xmlResponseCode, equalTo("Ok"));
+        Assertions.assertEquals(200, statusCode);
+
+    }
+
+    @Test
+    public void validateCustRefIssueWarrant() {
+
+        XmlPath xmlPath = new XmlPath(XMLFunctions.getCleanResponse(response)).setRoot("Envelope.Body.bulkResponse");
+        String customerReference = xmlPath.getString("customerReference");
+
+        assertThat(customerReference, equalTo(Integer.toString(randomNumber)));
+
+    }
+
+    @Test
+    public void validateSDTBulkRefIssueWarrant() {
+ 
+        XmlPath xmlPath = new XmlPath(XMLFunctions.getCleanResponse(response)).setRoot("Envelope.Body.bulkResponse");
+        String sdtBulkReference = xmlPath.getString("sdtBulkReference");
+    
+        assertThat(sdtBulkReference, matchesRegex(sdtBulkReferenceRegex));
+    }
+
+    @Test
+    public void validateCustID() {
+ 
+        custId = "12345678";
+        setUp();
+        
+        XmlPath xmlPath = new XmlPath(XMLFunctions.getCleanResponse(response)).setRoot("Envelope.Body.bulkResponse");
```

Comment:
Changing the custId class field value like this will cause problems with subsequent tests.  The custId value is never set back to the correct value afterwards, so the tests that run after will always fail with an invalid customer id error.

To prevent potential errors it might be an idea to make the following changes:
1.) Add randomNumber, custId and requestCount parameters to the soapRequestBody() method.
2.) Instead of calling setUp() here, call the soapRequestBody() method with the new customer id value, followed by soapResponse().
The advantage with this is that you won't need to remember to reset the custId field at the end of the test.  It also guards against the case when the test fails due to an exception before the custId field can be reset.